src/pages/access_review/AppSummary.js

```diff
@@ -50,11 +62,23 @@ export const AppSummary = hh(class AppSummary extends React.Component {
 
   async componentDidMount() {
     await this.generateRestrictions(this.props.consent.dataUse);
+    if (!isNil(this.props.researcherProfile) && !isNil(this.props.researcherProfile.institutionId)) {
+      await this.getInstitutionFromProfile();
+    }
   }
 
   render() {
     const { darInfo, accessElection, consent, researcherProfile } = this.props;
-    const { piName, profileName, institution, department, city, country } = researcherProfile;
+    const isThePiProp = find({propertyKey: 'isThePI'})(researcherProfile.researcherProperties);
+    const piNameProp = find({propertyKey: 'piName'})(researcherProfile.researcherProperties);
+    const piName = isNil(isThePiProp) ? "- -" : isThePiProp.propertyValue ?
+      researcherProfile.displayName : isNil(piNameProp) ? "xx" : piNameProp.propertyValue;
+    const departmentProp = find({propertyKey: 'department'})(researcherProfile.researcherProperties);
+    const department = isNil(departmentProp) ? "" : departmentProp.propertyValue;
+    const cityProp = find({propertyKey: 'city'})(researcherProfile.researcherProperties);
+    const city = isNil(cityProp) ? "" : cityProp.propertyValue;
+    const countryProp = find({propertyKey: 'country'})(researcherProfile.researcherProperties);
+    const country = isNil(countryProp) ? "" : countryProp.propertyValue;
```

Comment:
This is a case where we should have an extracted method to handle this since we're using the same strings as keys.